src/paperless_tesseract/parsers.py

```diff
@@ -47,13 +48,38 @@ def get_thumbnail(self):
         out_path = os.path.join(self.tempdir, "convert.png")
 
         # Run convert to get a decent thumbnail
-        run_convert(
-            self.CONVERT,
-            "-scale", "500x5000",
-            "-alpha", "remove",
-            "{}[0]".format(self.document_path),
-            out_path
-        )
+        try:
+            run_convert(
+                self.CONVERT,
+                "-scale", "500x5000",
+                "-alpha", "remove",
+                "{}[0]".format(self.document_path),
+                out_path
+            )
+        except ParseError:
+            # if convert fails, fall back to extracting
+            # the first PDF page as a PNG using Ghostscript
+            self.log(
+                "warning",
+                "Thumbnail generation with ImageMagick failed, "
+                "falling back to Ghostscript."
+            )
+            gs_out_path = os.path.join(self.tempdir, "gs_out.png")
+            cmd = [self.GHOSTSCRIPT,
+                   "-q",
+                   "-sDEVICE=pngalpha",
+                   "-o", gs_out_path,
+                   self.document_path]
+            if not subprocess.Popen(cmd).wait() == 0:
+                raise ParseError("Thumbnail (gs) failed at {}".format(cmd))
+            # then run convert on the output from gs
+            run_convert(
+                self.CONVERT,
+                "-scale", "500x5000",
```

Comment:
I'm not sure whether we should move this into a variable, as this is used in two places and one of them might be forgotten if it's ever changed. Could be done like this:
```python
from functools import partial

convert = partial(run_convert,
    self.CONVERT,
    "-scale", "500x5000",
    "-alpha", "remove")
```
and then use it like so
```python
convert("{}[0]".format(self.document_path), out_path)
```
and
```python
convert(gs_out_path, out_path)
```

@pitkley Any comments?

Reply:
My thoughts: 
I generally agree, but the two calls to `run_convert` are very close together - in one try-except block. And I would expect that anyone making (or reviewing) changes will at least read the whole function.
I have no problem doing it with `partial`, but to me the additional layer of abstraction makes it harder to understand at first glance what the code is doing.

Reply:
Fair enough.